myfitnesspal/client.py

```diff
@@ -804,3 +810,535 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+                            carbohydrates = round(carbohydrates / 4.184, 2)
+                            protein = round(protein / 4.184, 2)
+                            fat = round(fat / 4.184, 2)
+                    else:
+                        logger.warning("Provided percentage values are not plausible - Not able to update goals.")
+                        return False
+        else:
+            macro_energy = carbohydrates * 4 + protein * 4 + fat * 9
+            if energy_unit == "kilojoules":
+                macro_energy *= 4.184
+            # Compare energy values and set it correctly due to macros. Will also fix if no energy_value was provided.
+            if energy != macro_energy:
+                logger.warning(
+                    "Provided energy value and calculated energy value from macros do not match! Will override!"
+                )
+                energy = macro_energy
+
+        # Build payload based on observed browser behaviour
+        # TODO Insert additional micro nurtitions
+        new_goals = {}
+        new_goals["item"] = old_goals["items"][0]
+        new_goals["item"].pop("valid_to", None)
+        new_goals["item"].pop("default_group_id", None)
+        new_goals["item"].pop("updated_at", None)
+        new_goals["item"]["default_goal"]["meal_goals"] = []
+
+        # insert new values
+        new_goals["item"]["valid_from"] = today
+
+        new_goals["item"]["default_goal"]["energy"]["value"] = energy
+        new_goals["item"]["default_goal"]["energy"]["unit"] = energy_unit
+        new_goals["item"]["default_goal"]["carbohydrates"] = carbohydrates
+        new_goals["item"]["default_goal"]["protein"] = protein
+        new_goals["item"]["default_goal"]["fat"] = fat
+        # TODO Add micro nutritions
+
+        for goal in new_goals["item"]["daily_goals"]:
+            goal["meal_goals"] = []
+            goal.pop("group_id", None)
+
+            goal["energy"]["value"] = energy
+            goal["energy"]["unit"] = energy_unit
+            goal["carbohydrates"] = carbohydrates
+            goal["protein"] = protein
+            goal["fat"] = fat
+            # TODO Add micro nutritions
+
+        # Build Post-Request
+        # Post Request
+        url = parse.urljoin(self.BASE_API_URL, "v2/nutrient-goals")
+        result = self.session.post(url, json.dumps(new_goals), headers=auth_header)
+
+        # TODO Check Request Result
+        if result.status_code == 200:
+            return True
+        elif not result.ok:
+            logger.warning(
+                "Request Error - Unable to submit Goals to MyFitnessPal: "
+                "status code: {status}".format(status=result.status_code)
+            )
+            return False
+        else:
+            logger.error(
+                "Request Error - Unable to submit Goals to MyFitnessPal: "
+                "status code: {status}".format(status=result.status_code)
+            )
+            print(result)
+            return False
```

Comment:
```suggestion
        if not result.ok:
            raise SomeKindOfException(
                "Request Error - Unable to submit Goals to MyFitnessPal: "
                "status code: {status}".format(status=result.status_code)
            )
```

I think the above is pretty self-explanatory, but in case it isn't, I believe this is functionally essentially the same as your original implementation in all ways except how it responds to the non-OK status (by raising an exception) & doesn't return a `True` value.